Fix: day or year view render on event page

PratapRathi opened the pull request:
**What kind of change does this PR introduce?**
This PR will fix the rendering error on event page when select the year or day view.
We conditionally rendering the 'renderHours' function which has useMemo hook so react produce error "rendered more hooks than previous" so useMemo hook is removed from renderHour function.

<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #3279

**Snapshots/Videos:**

https://github.com/user-attachments/assets/c65a57d5-b9e8-4f0a-afb4-9d22efbc0879

**If relevant, did you update the documentation?**
N/A

**Summary**
**Does this PR introduce a breaking change?**
No

## Checklist

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced user interface for sorting options in the Event Header with clearer string labels.
	- Added a test case to verify the rendering of the hour view in the Event Calendar.

- **Refactor**
	- Simplified the logic for determining whether to show the "View more" button in the Event Calendar component by removing unnecessary memoization.
	- Streamlined the JSX structure in the Calendar component for improved readability. 

- **Bug Fixes**
	- Updated test cases to ensure correct interaction simulation with the Event Header component.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

palisadoes reviewed (changes requested):
1. This is not fixed. 
2. Though the screen is rendered. None of the events for that day are listed.
3. Additionally
![image](https://github.com/user-attachments/assets/08419940-dc5e-4a43-a56c-17bdcf4f034c)

palisadoes reviewed (changes requested):
Please add tests to ensure this doesn't happen again.

PratapRathi commented:
1. Scrollbar is removed
2. added new test case for `day view`.
3. removed duplicate component of day and year view
4. Event are listed in day view

**Earlier- duplicate element visible on day and calendar view**

<img width="1462" alt="Screenshot 2025-01-22 at 11 00 51" src="https://github.com/user-attachments/assets/f9cbf8a2-4155-4eeb-be52-28d5d9818495" />

**Now**
[Screen_recording](https://github.com/user-attachments/assets/4be5ad48-7406-47b6-ba25-a0d1bfdb482f)

palisadoes commented:
Everything in this PR

palisadoes commented:
Make it another issue in second thought
